src/System.Private.CoreLib/shared/System/SpanHelpers.Byte.cs

```diff
@@ -1107,23 +1108,37 @@ private static int LocateLastFoundByte(Vector<byte> match)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static int LocateFirstFoundByte(ulong match)
         {
-            // Flag least significant power of two bit
-            var powerOfTwoFlag = match ^ (match - 1);
-            // Shift all powers of two into the high byte and extract
-            return (int)((powerOfTwoFlag * XorPowerOfTwoToHighByte) >> 57);
+            if (Bmi1.IsSupported && IntPtr.Size == 8)
+            {
+                return (int)(Bmi1.TrailingZeroCount(match) >> 3);
```

Comment:
Recently, we have changed this API to `Bmi1.X64.TrailingZeroCount(ulong)` https://github.com/dotnet/coreclr/pull/20146

Next week, I will implement all the new 64-bit-only intrinsic (exposed in `X64`) in JIT and remove the legacy APIs. So, I suggest to wait a bit and use the new APIs.

BTW, with the new 64-bit-only intrinsic, you do not need `Bmi1.IsSupported && IntPtr.Size == 8`, just `Bmi1.X64.IsSupported`.

Reply:
Marked as WIP then